lib/SILOptimizer/Mandatory/YieldOnceCheck.cpp

```diff
@@ -392,24 +401,24 @@ class YieldOnceCheck : public SILFunctionTransform {
 
       // Find all transitive predecessors of 'yieldPred' that were visited
       // during the analysis
-      SmallPtrSet<SILBasicBlock *, 8> predecessorsOfYieldPred;
+      BasicBlockSet predecessorsOfYieldPred(&fun);
       for (auto *predBB :
            llvm::breadth_first<llvm::Inverse<SILBasicBlock *>>(yieldPred)) {
-        if (visitedBBs.count(predBB)) {
+        if (visitedBBs[predBB].isVisited()) {
```

Comment:
One thing that I am noticing in this change is that a bunch of const code is being replaced with non-const code like this. My worry is that we are relying on BasicBlockData to initialize data for all blocks even if we are only going to visit a few.

That could change the algorithmic behavior to change, no? Is it possible to make BasicBlockData initialize by default in a lazy way? That might help. But I am still worried about mallocing for all blocks in a function. @eeckstein your thoughts?

Reply:
I talked with eeckstein offline and he said that he only changed to use BasicBlockData for cases where we have to visit the whole CFG, so I'm cool with this.

Reply:
In case of YieldOnceCheck the "visitedBBs" data is only created once a run of the pass. And BasicBlockData should only be used for such cases.
Yes, we could make an extended version of BasicBlockData which lazily constructs data (by using it together with BasicBlockFlag). That's future work.